modules/user/src/main/DataForm.scala

```diff
@@ -15,6 +15,15 @@ final class DataForm(authenticator: Authenticator) {
 
   case class NoteData(text: String, mod: Boolean)
 
+  def username(user: User): Form[UserNameChange] = Form(mapping(
+    "userName" -> text.verifying("usernameNotSame", name =>
+      name.toLowerCase == user.username.toLowerCase && name != user.username)
+  )(UserNameChange.apply)(UserNameChange.unapply))
+
+  case class UserNameChange(userName: String)
```

Comment:
`case class UsernameChange(username: String)` since lila uses `username` and not `userName`.